jvm/sargon-android/src/main/java/com/radixdlt/sargon/os/driver/AndroidBiometricAuthorizationDriver.kt

```diff
@@ -87,24 +48,24 @@ class BiometricsHandler(
     private val biometricsResultsChannel = Channel<Result<Unit>>()
 
     fun register(activity: FragmentActivity) {
-        activity.lifecycleScope.launch {
-            // Listen to biometric prompt requests while the activity is at least started.
-            activity.lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
-                biometricRequestsChannel.receiveAsFlow().collectLatest {
-                    val result = requestBiometricsAuthorization(activity)
-
-                    // Send back the result to sargon os
-                    biometricsResultsChannel.send(result)
-                }
+        biometricRequestsChannel
+            .receiveAsFlow()
```

Comment:
why you define the `biometricRequestsChannel` and `biometricsResultsChannel` as `Channel` and then receive the as `Flow`? 
Why not to define them as Flows?

Reply:
I wanted to implement a logic of two cold flows. Sargon os can ask for biometrics at any time. So I wanted to solve 2 problems

1. No activity is yet `STARTED` in order to handle biometrics. Remember in order to ask for biometrics, we need a reference to a `FragmentActivity`. In order for the biometrics dialog to show up, we need the activity to be at least started. So I wanted to have a suspending function waiting for an activity to either register and start or return from background. This is why `biometricRequestsChannel` exists. It waits for the request to be handled by one activity.
2. The `biometricsResultsChannel` is waiting for the **USER** to respond. The values of the channel will be emitted by another function.